{
  "artifacts": [],
  "command": "main",
  "experiment": {
    "base_dir": "/home/gangus/cow-tus/cow_tus/experiment",
    "dependencies": [
      "emmental==0.0.4",
      "numpy==1.17.4",
      "Pillow==6.2.1",
      "sacred==0.8.0",
      "scipy==1.3.2",
      "torch==1.3.1",
      "torchvision==0.4.2"
    ],
    "mainfile": "train_emmental.py",
    "name": "trainer",
    "repositories": [
      {
        "commit": "0d9f86fd9a0f5de3729a9550fb8b1cbca08a63cb",
        "dirty": true,
        "url": "https://code.stanford.edu/lungrenlab/cow-tus.git/"
      },
      {
        "commit": "0d9f86fd9a0f5de3729a9550fb8b1cbca08a63cb",
        "dirty": true,
        "url": "https://code.stanford.edu/lungrenlab/cow-tus.git/"
      }
    ],
    "sources": [
      [
        "../data/transforms.py",
        "_sources/transforms_ca8e5ead7f57a284878261d9c7b63666.py"
      ],
      [
        "train_emmental.py",
        "_sources/train_emmental_f90a8654e6a9b6fb546ae80e5abc36c1.py"
      ]
    ]
  },
  "fail_trace": [
    "Traceback (most recent call last):\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/sacred/stdout_capturing.py\", line 163, in tee_output_fd\n    yield out  # let the caller do their printing\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/sacred/run.py\", line 238, in __call__\n    self.result = self.main_function(*args)\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/sacred/config/captured_function.py\", line 42, in captured_function\n    result = wrapped(*args, **kwargs)\n",
    "  File \"cow_tus/experiment/train_emmental.py\", line 230, in main\n    res = trainer.run()\n",
    "  File \"cow_tus/experiment/train_emmental.py\", line 217, in run\n    learner.learn(self.model, self.dataloaders)\n",
    "  File \"/home/gangus/cow-tus/.emmental/src/emmental/learner.py\", line 478, in learn\n    uids, X_dict, Y_dict, task_to_label_dict\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/modules/module.py\", line 541, in __call__\n    result = self.forward(*input, **kwargs)\n",
    "  File \"/home/gangus/cow-tus/.emmental/src/emmental/model.py\", line 261, in forward\n    output_dict = self.flow(X_dict, list(task_to_label_dict.keys()))\n",
    "  File \"/home/gangus/cow-tus/.emmental/src/emmental/model.py\", line 223, in flow\n    output = self.module_pool[action[\"module\"]].forward(*input)\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/parallel/data_parallel.py\", line 152, in forward\n    outputs = self.parallel_apply(replicas, inputs, kwargs)\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/parallel/data_parallel.py\", line 162, in parallel_apply\n    return parallel_apply(replicas, inputs, kwargs, self.device_ids[:len(replicas)])\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/parallel/parallel_apply.py\", line 85, in parallel_apply\n    output.reraise()\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/_utils.py\", line 385, in reraise\n    raise self.exc_type(msg)\n",
    "RuntimeError: Caught RuntimeError in replica 0 on device 0.\nOriginal Traceback (most recent call last):\n  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/parallel/parallel_apply.py\", line 60, in _worker\n    output = module(*input, **kwargs)\n  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/modules/module.py\", line 541, in __call__\n    result = self.forward(*input, **kwargs)\n  File \"/home/gangus/cow-tus/cow_tus/models/modules/encoders.py\", line 37, in forward\n    out = self.maxPool3d_2a_3x3(out)\n  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/modules/module.py\", line 541, in __call__\n    result = self.forward(*input, **kwargs)\n  File \"/home/gangus/cow-tus/cow_tus/models/modules/i3d.py\", line 122, in forward\n    out = self.pool(inp)\n  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/modules/module.py\", line 541, in __call__\n    result = self.forward(*input, **kwargs)\n  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/modules/pooling.py\", line 210, in forward\n    self.return_indices)\n  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/_jit_internal.py\", line 138, in fn\n    return if_false(*args, **kwargs)\n  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/torch/nn/functional.py\", line 520, in _max_pool3d\n    input, kernel_size, stride, padding, dilation, ceil_mode)\nRuntimeError: CUDA out of memory. Tried to allocate 736.00 MiB (GPU 0; 11.78 GiB total capacity; 6.55 GiB already allocated; 465.62 MiB free; 13.87 MiB cached)\n\n",
    "\nDuring handling of the above exception, another exception occurred:\n\n",
    "Traceback (most recent call last):\n",
    "  File \"/usr/lib/python3.7/contextlib.py\", line 130, in __exit__\n    self.gen.throw(type, value, traceback)\n",
    "  File \"/home/gangus/cow-tus/.env/lib/python3.7/site-packages/sacred/stdout_capturing.py\", line 175, in tee_output_fd\n    tee_stdout.wait(timeout=1)\n",
    "  File \"/usr/lib/python3.7/subprocess.py\", line 990, in wait\n    return self._wait(timeout=timeout)\n",
    "  File \"/usr/lib/python3.7/subprocess.py\", line 1616, in _wait\n    raise TimeoutExpired(self.args, timeout)\n",
    "subprocess.TimeoutExpired: Command '['tee', '-a', '/tmp/tmpg48e9pf5']' timed out after 1 seconds\n"
  ],
  "heartbeat": "2019-11-19T17:53:07.652246",
  "host": {
    "ENV": {},
    "cpu": "Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz",
    "gpus": {
      "driver_version": "430.34",
      "gpus": [
        {
          "model": "TITAN V",
          "persistence_mode": false,
          "total_memory": 12062
        },
        {
          "model": "TITAN V",
          "persistence_mode": false,
          "total_memory": 12066
        },
        {
          "model": "TITAN V",
          "persistence_mode": false,
          "total_memory": 12066
        },
        {
          "model": "TITAN V",
          "persistence_mode": false,
          "total_memory": 12066
        }
      ]
    },
    "hostname": "u07082019000",
    "os": [
      "Linux",
      "Linux-4.15.0-55-generic-x86_64-with-Ubuntu-18.04-bionic"
    ],
    "python_version": "3.7.4"
  },
  "meta": {
    "command": "main",
    "comment": "this recreates the full size, modified binary experiment on representative loops. This time, we validate with the appropriate set",
    "options": {
      "--beat-interval": null,
      "--capture": null,
      "--comment": "this recreates the full size, modified binary experiment on representative loops. This time, we validate with the appropriate set",
      "--debug": false,
      "--enforce_clean": false,
      "--file_storage": null,
      "--force": false,
      "--help": false,
      "--loglevel": null,
      "--mongo_db": null,
      "--name": null,
      "--pdb": false,
      "--print-config": true,
      "--priority": null,
      "--queue": false,
      "--s3": null,
      "--sql": null,
      "--tiny_db": null,
      "--unobserve": false,
      "COMMAND": null,
      "UPDATE": [],
      "help": false,
      "with": false
    }
  },
  "resources": [],
  "result": null,
  "start_time": "2019-11-19T17:52:29.424257",
  "status": "FAILED",
  "stop_time": "2019-11-19T17:53:07.655075"
}